### DISCUSSION TITLE
`ubuntu-latest` and `macos-latest` is not the latest

### DISCUSSION THREAD

**Discussion by kwaa** [community/community#29361] [Actions] [answered]
https://docs.github.com/en/actions/using-github-hosted-runners/about-github-hosted-runners#supported-runners-and-hardware-resources

It's very confusing.
when I fill in `ubuntu-latest` in yaml,
it means I want to choose the latest OS that I can use in GitHub Actions,
rather than an older version.

this reminds me of a year or two ago,
when the latest version was `ubuntu-20.04` and the "latest" version was 18.04...

**Comment by juliandunn** [comment, marked as answer]
The meaning of `latest` is documented here: https://github.com/actions/runner-images#label-scheme

It is not intended to mean the latest version of whatever OS is available in the world, because we need time to react to those new releases, validate & test our images, and give sufficient notice to customers before moving the `latest` label to that version.

**Reply by kwaa** [reply]
Well... I still don't quite understand it. Since the new version of the system is already available in GitHub Actions (and can be considered verified and tested), it should be "latest". People who don't want the system to stay up to date should specify a numeric version number and modify it to upgrade if needed, no?

**Reply by juliandunn** [reply]
That's not generally how we find people want to use the system:

* They want to set their `runs-on:` once and not have to bulk-modify hundreds of workflows whenever a new version of Ubuntu comes out, and
* A large proportion of CI jobs are pretty OS-version independent, e.g.. they are unlikely to break whenever the pointer of `ubuntu-latest` moves from 18.04 to 20.04.

The meaning of `latest` today is essentially "latest stable". But your feedback is valuable. Maybe in the future we'll consider just calling it something like `ubuntu-stable` or `ubuntu-ga`, as you are right, the word "latest" is very much up for interpretation semantically.

**Reply by kwaa** [reply]
> That's not generally how we find people want to use the system:
> 
> * They want to set their `runs-on:` once and not have to bulk-modify hundreds of workflows whenever a new version of Ubuntu comes out, and
> * A large proportion of CI jobs are pretty OS-version independent, e.g.. they are unlikely to break whenever the pointer of `ubuntu-latest` moves from 18.04 to 20.04.
> 
> The meaning of `latest` today is essentially "latest stable". But your feedback is valuable. Maybe in the future we'll consider just calling it something like `ubuntu-stable` or `ubuntu-ga`, as you are right, the word "latest" is very much up for interpretation semantically.

I also think it would be more appropriate to rename `latest` to `stable`.
Perhaps a true latest version option could also be provided?
